app/routes/claim/view-claim.js

```diff
@@ -49,26 +35,23 @@ module.exports = function (router) {
         if (updateConflict) {
           throw new ValidationError({UpdateConflict: [ValidationErrorMessages.getUpdateConflict(lastUpdatedData.Status)]})
         }
-        var claimExpenses = getClaimExpenseResponses(req.body)
-        var claimDecision = new ClaimDecision(
-          req.user.email,
-          req.body.assistedDigitalCaseworker,
-          req.body.decision,
-          req.body.reasonRequest,
-          req.body.reasonReject,
-          req.body.additionalInfoApprove,
-          req.body.additionalInfoRequest,
-          req.body.additionalInfoReject,
-          req.body.nomisCheck,
-          req.body.dwpCheck,
-          req.body.visitConfirmationCheck,
-          claimExpenses
-        )
-
-        SubmitClaimResponse(req.params.claimId, claimDecision)
-          .then(function () {
-            return res.redirect('/')
-          })
+
+        var removeDeductionClicked = false
+
+        Object.keys(req.body).forEach(function (key) {
```

Comment:
should not be necessary to search key, should check by index `req.body['remove-deduction']`.
If not move to function